src/vmm/src/devices/virtio/transport/mod.rs

```diff
@@ -1,5 +1,37 @@
 // Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::sync::Arc;
+use std::sync::atomic::AtomicU32;
+
+use vmm_sys_util::eventfd::EventFd;
+
 /// MMIO transport for VirtIO devices
 pub mod mmio;
+
+/// Represents the types of interrupts used by VirtIO devices
+#[derive(Debug, Clone)]
+pub enum VirtioInterruptType {
+    /// Interrupt for VirtIO configuration changes
+    Config,
+    /// Interrupts for new events in a queue.
+    Queue(u16),
+}
+
+/// API of interrupt types used by VirtIO devices
+pub trait VirtioInterrupt: std::fmt::Debug + Send + Sync {
```

Comment:
Mh, what is the advantage of this being a trait over it being an enum?

Reply:
Don't necessarily see an advantage. That's how it's defined on the CH side, so got it as is.
I do have a sort of personal preference to dynamic dispatch when it's not on a performance critical path, because I (visually) hate this:

```Rust
impl VirtioInterrupt {
    pub fn foo(&self) -> Bar {
        match self {
             Irq(irq) => irq.foo()
             Msi(msi) => msi.foo()
        }
    }
}
```

Reply:
Ah fair, I personally actually prefer the enum version, but that's indeed personal preference. I mainly left the comment because I found the explicit Debug bound on a trait that functionally has nothing to do with debugging a bit ugly, but can also stay like this if you prefer it